modules/core/src/main/kotlin/com/github/l34130/mise/toolWindow/MisePanel.kt

```diff
@@ -0,0 +1,376 @@
+import com.intellij.notification.NotificationType
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.actionSystem.*
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.openapi.options.ShowSettingsUtil
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.wm.ToolWindowManager
+import com.intellij.ui.PopupHandler
+import com.intellij.ui.components.JBPanel
+import com.intellij.ui.components.JBScrollPane
+import com.intellij.ui.treeStructure.Tree
+import com.intellij.util.ui.JBUI
+import com.intellij.util.ui.UIUtil
+import org.jetbrains.plugins.terminal.TerminalView
+import java.awt.BorderLayout
+import java.awt.Component
+import java.awt.event.MouseEvent
+import javax.swing.JTree
+import javax.swing.UIManager
+import javax.swing.tree.DefaultMutableTreeNode
+import javax.swing.tree.DefaultTreeCellRenderer
+import javax.swing.tree.DefaultTreeModel
+import javax.swing.tree.TreePath
+
+class RefreshAction(private val panel: MisePanel) : AnAction(
+    "Refresh Mise Configuration",
+    "Refresh Mise configuration",
+    AllIcons.Actions.Refresh
+) {
+    override fun actionPerformed(e: AnActionEvent) {
+        FileDocumentManager.getInstance().saveAllDocuments()
+        panel.refreshMiseConfiguration()
+    }
+}
+
+class SettingsAction(private val project: Project) : AnAction(
+    "Mise Settings",
+    "Open Mise settings",
+    AllIcons.General.Settings
+) {
+    override fun actionPerformed(e: AnActionEvent) {
+        ShowSettingsUtil.getInstance().showSettingsDialog(project, MiseConfigurable::class.java)
+    }
+}
+
+class MisePanel(private val project: Project) : JBPanel<JBPanel<*>>(BorderLayout()), Disposable {
+    private val rootNode = DefaultMutableTreeNode("Root")
+    private val treeModel = DefaultTreeModel(rootNode)
+    private val tree = Tree(treeModel)
+    private val settingsConnection = project.messageBus.connect(this)
+
+    init {
+        tree.apply {
+            isRootVisible = false
+            showsRootHandles = true
+            cellRenderer = MiseTreeCellRenderer()
+            background = UIManager.getColor("Tree.background")
+            isOpaque = false
+            addMouseListener(TreeMouseListener())
+        }
+
+        add(JBScrollPane(tree).apply {
+            border = JBUI.Borders.empty()
+            viewport.isOpaque = false
+            background = UIManager.getColor("Tree.background")
+            isOpaque = false
+        }, BorderLayout.CENTER)
+
+        refreshMiseConfiguration()
+
+        settingsConnection.subscribe(
+            MiseSettings.MISE_SETTINGS_TOPIC,
+            object : MiseSettings.SettingsChangeListener {
+                override fun settingsChanged(oldState: MiseSettings.State, newState: MiseSettings.State) {
+                    if (oldState.miseProfile != newState.miseProfile) {
+                        refreshMiseConfiguration()
+                    }
+                }
+            }
+        )
```

Comment:
Some registering logic maybe better to be seperated.

Reply:
Feel free to update. I was just trying to have some sync with the panel, but it's not working correctly 😅